### PR TITLE
Add parameter values option to execute api

### PR DESCRIPTION
**Pull request by YannanGao-gs**
#### What type of PR is this?
Feature Request

#### What does this PR do / why is it needed ?
In Studio, We want to remove the usage of let statements when executing queries with parameters.
The use of let statements makes it difficult for engine to do type matching causing bugs when executing queries with parameters.
As a result we should move over and use engine to execute the queries with parameters.

#### Which issue(s) this PR fixes:

Fixes https://github.com/finos/legend-engine/issues/1018
Links to https://github.com/finos/legend-studio/issues/1535

### CODE REVIEW

File: legend-engine-query-pure/src/main/java/org/finos/legend/engine/query/pure/api/Execute.java [hunk 1]
```diff
@@ -357,10 +415,26 @@ private Response executeAsMiddleTierPlan(PlanExecutor planExecutor, MutableList<
                 )
         );
 
+        MutableMap<String, Result> vars = Maps.mutable.empty();
+        List<Variable> functionParameters = ((SingleExecutionPlan) plan).rootExecutionNode.executionNodes.stream().filter(node -> node instanceof FunctionParametersValidationNode).map(executionNode -> ((FunctionParametersValidationNode) executionNode).functionParameters).flatMap(Collection::stream).collect(Collectors.toList());
+        Multiplicity paraMultiplicity = new Multiplicity();
+        for (Map.Entry<String, ?> e : parameters.entrySet())
+        {
+            paraMultiplicity = getFunctionParameterMultiplicity(functionParameters, paraMultiplicity, e);
```
Comment (MauricioUyaguari) [inline legend-engine-query-pure/src/main/java/org/finos/legend/engine/query/pure/api/Execute.java:423]: lets add `parameterValue = parameterToValue.getValue()` before this and use it below to make code more readible

**Review by MauricioUyaguari — CHANGES_REQUESTED** [review]
Looks good. Just small cleanup stuff on naming. 
Also lets add an api test to test execution with executeInput with parameterValues filled out.

File: legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/api/model/ParameterValue.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+public class ParameterValue
+{
+    @JsonProperty
+    public String name;
+    @JsonProperty
+    public ValueSpecification value;
+
+    @JsonCreator
+    public ParameterValue(@JsonProperty("name") String name, @JsonProperty("value") ValueSpecification value)
```
Comment (Yasirmod17) [inline legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/api/model/ParameterValue.java:29]: use existing class. Move to proper module if necessary

File: legend-engine-testable/src/main/java/org/finos/legend/engine/testable/helper/PrimitiveValueSpecificationToObjectVisitor.java [hunk 3]
```diff
@@ -123,7 +123,7 @@ public Object visit(Enum anEnum)
     @Override
     public Object visit(EnumValue enumValue)
     {
-        throw new UnsupportedOperationException("Unsupported value specification type");
+        return enumValue.value;
```
Comment (Yasirmod17) [inline legend-engine-testable/src/main/java/org/finos/legend/engine/testable/helper/PrimitiveValueSpecificationToObjectVisitor.java:126]: Add an enum test to trigger this flow

File: legend-engine-query-pure/src/test/java/org/finos/legend/engine/query/pure/api/test/inMemory/TestServiceExecutionWithParameters.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        ModelManager modelManager = new ModelManager(DeploymentMode.TEST);
+        RelationalStoreExecutor relationalStoreExecutor = new RelationalStoreExecutorBuilder().build();
+        PlanExecutor planExecutor = PlanExecutor.newPlanExecutor(relationalStoreExecutor);
+        HttpServletRequest request = (HttpServletRequest) Proxy.newProxyInstance(getClass().getClassLoader(), new java.lang.Class<?>[] {HttpServletRequest.class}, new TestServiceExecutionWithParameters.ReflectiveInvocationHandler(new TestServiceExecutionWithParameters.Request()));
+        Response result = new Execute(modelManager, planExecutor, (PureModel pureModel) -> Root_meta_relational_extension_relationalExtensions__Extension_MANY_(pureModel.getExecutionSupport()), LegendPlanTransformers.transformers).execute(request, input, SerializationFormat.defaultFormat, null, null);
+        Assert.assertEquals(200, result.getStatus());
+        return result;
+    }
+
+    private BaseExecutionContext context()
```
Comment (Yasirmod17) [inline legend-engine-query-pure/src/test/java/org/finos/legend/engine/query/pure/api/test/inMemory/TestServiceExecutionWithParameters.java:77]: Move to helper and merge with other tests

**Review by Yasirmod17 — CHANGES_REQUESTED** [review]

**Review by Yasirmod17 — APPROVED** [review]

**Review by MauricioUyaguari — APPROVED** [review]

**Review by Yasirmod17 — APPROVED** [review]

**Pull request merged**